packages/thirdweb/src/extensions/farcaster/write/addSigner.ts

```diff
@@ -0,0 +1,119 @@
+import { toBigInt } from "../../../utils/bigint.js";
+import { prepareContractCall } from "../../../transaction/prepare-contract-call.js";
+import type { ThirdwebClient } from "../../../client/client.js";
+import type { Account } from "../../../wallets/interfaces/wallet.js";
+import type { Chain } from "../../../chains/types.js";
+import { getKeyGateway } from "../contracts.js";
+import type { Hex } from "../../../utils/encoding/hex.js";
+import type { Prettify } from "../../../utils/type-utils.js";
+import type { Address } from "abitype";
+
+/**
+ * Represents the parameters for the `addSigner` function.
+ *
+ * @remarks
+ *
+ * This function can be used wither be provided pre-generated signatures or the wallet accounts directly.
+ * This is done so the helpers can be used when there's no direct access to the account, but signatures can be generated (e.g. engine)
+ *
+ * If the `appAccount` is provided, the `signedKeyRequestMetadata`, `appAccountAddress` and `deadline` must be provided.
+ * `deadline` must match the one used to generate the signature.
+ */
+export type AddSignerParams = Prettify<
+  {
+    client: ThirdwebClient;
+    signerPublicKey: Hex;
+    chain?: Chain;
+    disableCache?: boolean;
+  } & (
+    | {
+        appAccount: Account;
+      }
+    | {
+        signedKeyRequestMetadata: Hex;
+        appAccountAddress: Address;
+        deadline: bigint;
+      }
+  )
+>;
+
+/**
+ * Adds farcaster signer for the given account.
+ * @param options - The options for adding the signer.
+ * @returns A prepared transaction object to add the signer to the account.
+ * @extension FARCASTER
+ * @example
+ * ```ts
+ * import { addSigner } from "thirdweb/extensions/farcaster";
+ * const tx = await addSigner({
+ *  client,
+ * 	appAccount,
+ *  signerPublicKey
+ * });
+ * ```
+ */
+export function addSigner(options: AddSignerParams) {
+  return prepareContractCall({
```

Comment:
@jnsdls we should do the thing we talked about where you can pass the args OR a async fn that returns the args on generated extensions

Reply:
yeah, I'll play with this today

Reply:
have not gotten to finish this yet, but working on it tonight